lib/os_places_api/client.rb

```diff
@@ -15,15 +15,18 @@ def locations_for_postcode(postcode)
       response = get_api_response(postcode)
       raise NoResultsForPostcode if response["results"].nil?
 
-      Postcode.create!(postcode: postcode, results: response["results"]) unless Postcode.find_by(postcode: postcode)
+      if any_locations?(response["results"]) && !Postcode.find_by(postcode: postcode)
+        Postcode.create!(postcode: postcode, results: response["results"])
+      end
       build_locations(response["results"])
     end
 
     def update_postcode(postcode)
       postcode = PostcodeHelper.normalise(postcode)
       response = get_api_response(postcode)
       record = Postcode.find_by(postcode: postcode)
-      if response["results"].nil?
+
+      if response["results"].nil? || !any_locations?(response["results"])
```

Comment:
I did initially think that deleting the record should not be in scope for this (we could run a one-off rake task instead) but I also like the 'self healing' element of this, and I guess it covers us if a previously 'valid' postcode becomes filtered.